src/pages-and-resources/discussions/app-list/AppList.test.jsx

```diff
@@ -20,84 +19,109 @@ import { fetchDiscussionSettings, fetchProviders } from '../data/thunks';
 import {
   generateProvidersApiResponse,
   piazzaApiResponse,
+  legacyApiResponse,
 } from '../factories/mockApiResponses';
 import AppList from './AppList';
 import messages from './messages';
 
 const courseId = 'course-v1:edX+TestX+Test_Course';
+let axiosMock;
+let store;
+let container;
+
+const mockStore = async (mockResponse, provider) => {
+  axiosMock.onGet(getDiscussionsProvidersUrl(courseId))
+    .reply(200, generateProvidersApiResponse(false, provider));
+  axiosMock.onGet(getDiscussionsSettingsUrl(courseId)).reply(200, mockResponse);
+  await executeThunk(fetchProviders(courseId), store.dispatch);
+  await executeThunk(fetchDiscussionSettings(courseId), store.dispatch);
+};
+
+function renderComponent(screenWidth = breakpoints.extraLarge.minWidth) {
+  const wrapper = render(
+    <AppProvider store={store}>
+      <ResponsiveContext.Provider value={{ width: screenWidth }}>
+        <IntlProvider locale="en" messages={{}}>
+          <AppList />
+        </IntlProvider>
+      </ResponsiveContext.Provider>
+    </AppProvider>,
+  );
+  container = wrapper.container;
+}
 
 describe('AppList', () => {
-  let axiosMock;
-  let store;
-  let container;
+  describe('AppList for Admin role', () => {
+    beforeEach(async () => {
+      initializeMockApp({
+        authenticatedUser: {
+          userId: 3,
+          username: 'abc123',
+          administrator: true,
+          roles: [],
+        },
+      });
 
-  function createComponent(screenWidth = breakpoints.extraLarge.minWidth) {
-    return (
-      <AppProvider store={store}>
-        <ResponsiveContext.Provider value={{ width: screenWidth }}>
-          <IntlProvider locale="en" messages={{}}>
-            <AppList />
-          </IntlProvider>
-        </ResponsiveContext.Provider>
-      </AppProvider>
-    );
-  }
+      store = await initializeStore();
+      axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+      await mockStore(piazzaApiResponse);
+    });
 
-  beforeEach(async () => {
-    initializeMockApp({
-      authenticatedUser: {
-        userId: 3,
-        username: 'abc123',
-        administrator: true,
-        roles: [],
-      },
+    test('display a card for each available app', async () => {
+      renderComponent();
+      const appCount = store.getState().discussions.appIds.length;
+      expect(screen.queryAllByRole('radio')).toHaveLength(appCount);
```

Comment:
You can directly pass the length instead of creating a variable appCount. It's recommended to avoid declaring variables that are only used once for the sake of cleaner and more concise code.

expect(screen.queryAllByRole('radio')).toHaveLength(store.getState().discussions.appIds.length);

Reply:
`appCount` is used for code readability here